src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,39 @@
+package mate.academy.bookstore.repository.impl;
+
+import java.util.List;
+import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.repository.BookRepository;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```

Comment:
hint: you can use lombok's `@RequiredArgsConstructor` and remove this at all

`@Autowired` annotation is optional on constructors, since Spring will automatically consider the constructor as a way to inject dependencies (**if you have just 1 constructor**)